buildenv/jenkins/pipeline-functions

```diff
@@ -51,6 +42,16 @@
     }
 }
 
+def get_sha(REPO, BRANCH) {
+    // Get the SHA at the tip of the BRANCH in REPO.
+    // Allows Pipelines to kick off multiple builds and have the same SHA built everywhere.
+    return sh (
+            // "git ls-remote $REPO" will return all refs, adding "$BRANCH" will only return the specific branch we are interested in
+            script: "git ls-remote $REPO refs/heads/$BRANCH | awk \'{ print \$1 }\' | sed -r 's/(^.{7}).*/\\1/'",
```

Comment:
Why the change to only return the first 7 characters of the SHA?

The `sed` filter eliminates the value of the `awk` part of the pipeline.

Reply:
The only thing I changed in the commands was removing the grep and adding the branch name to the git command. I agree we should just use the cut and eliminate the awk, will update.